nemo/collections/asr/models/pt_models.py

```diff
@@ -0,0 +1,288 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from math import ceil
+from typing import Dict, Optional, Union
+
+import torch
+from omegaconf import DictConfig, open_dict
+from pytorch_lightning import Trainer
+
+from nemo.collections.asr.data import audio_to_text_dataset
+from nemo.collections.asr.models.asr_model import ExportableEncDecModel
+from nemo.collections.asr.parts.mixins import ASRModuleMixin
+from nemo.collections.asr.parts.preprocessing.perturb import process_augmentations
+from nemo.core.classes import ModelPT
+from nemo.core.classes.common import PretrainedModelInfo, typecheck
+from nemo.core.neural_types import AudioSignal, LengthsType, NeuralType, SpectrogramType, VoidType
+from nemo.utils import logging
+
+__all__ = ['ASREncDecPTModel']
+
+
+class ASREncDecPTModel(ModelPT, ExportableEncDecModel, ASRModuleMixin):
```

Comment:
What does PT refer to here? PyTorch? Pre-Train? Expand its name.

Also, why is it inhering ModelPT? Nearly 70% of the code other than the forward and *_step() is same as EncDecCTCModel. Please inherit from it and override the necessary methods.

The dataloader steps are same as in EncDecCTCModels, same with input/output types. If needed you can disable a few methods such as transcribe() and list_pretrained_models() by raising NotImplementedError but it would cut down on dataset loading duplication

Reply:
Changed name (including ASR->Speech, since could potentially be used for other tasks too), added type annotations